### PR TITLE
Add `deploy_contract` tests

### PR DESCRIPTION
**Pull request by karol-bisztyga**

### CODE REVIEW

**Review by piotmag769 — COMMENTED** [review]
Shouldn't we test deploying cairo0 contract with constructor?
I would also test (for cairo0 and cairo1) deploying contracts by path.

https://github.com/software-mansion/protostar/pull/1561/files#diff-5a6190b786de870048cc98a39ef4fbf9b20261b0ecf5f2a43656253bcc8910d7

File: tests/integration/cairo1_hint_locals/deploy_contract/data/with_ctor_panic.cairo
[the remark this answers is not in the document]
Comment (Arcticae) [reply tests/integration/cairo1_hint_locals/deploy_contract/data/with_ctor_panic.cairo:1]: won't fit, too long of a name

**Review by Arcticae — APPROVED** [review]

**Comment by Arcticae** [comment]
@piotmag769 we don't support deploying by path anymore 
@karol-bisztyga i agree though on the  cairo0 contract with constructor test, can this be added before merging? 🥺

File: tests/integration/cairo1_hint_locals/deploy_contract/deploy_contract_test.cairo [hunk 1]
```diff
@@ -3,32 +3,32 @@ use result::ResultTrait;
 
 #[test]
 fn test_deploy_contract_simple() {
```
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply tests/integration/cairo1_hint_locals/deploy_contract/deploy_contract_test.cairo]: [fix name](https://github.com/software-mansion/protostar/pull/1732/commits/e1118c6156ce1312df80a09ef9bf8dfb75694b2b)

**Pull request merged**